test/new-e2e/tests/agent-metric-logs/log-agent/linux-log/file-tailing/utf_test.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package filetailing
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+)
+
+//go:embed log-config/utf-16-le.yaml
+var utfLittleEndianLogConfig []byte
+
+//go:embed log-config/utf-16-be.yaml
+var utfBigEndianLogConfig []byte
+
+const service = "utfservice"
+
+// UTFSuite defines a test suite for the log agent tailing UTF encoded logs
+type UtfSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+// TestUtfSuite runs the E2E test suite for tailing UTF encoded logs
+func TestUtfSuite(t *testing.T) {
+	s := &UtfSuite{}
+	devModeEnv, _ := os.LookupEnv("E2E_DEVMODE")
+	options := []e2e.SuiteOption{
+		e2e.WithProvisioner(awshost.Provisioner(awshost.WithAgentOptions(agentparams.WithLogs(), agentparams.WithIntegration("custom_logs.d", logConfig)))),
+	}
+	if devMode, err := strconv.ParseBool(devModeEnv); err == nil && devMode {
+		options = append(options, e2e.WithDevMode())
+	}
+
+	e2e.Run(t, s, options...)
+}
+
+func (s *UtfSuite) BeforeTest(suiteName, testName string) {
+	s.BaseSuite.BeforeTest(suiteName, testName)
+	// flush intake
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		err := s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+		if assert.NoErrorf(c, err, "Having issues flushing server and resetting aggregators, retrying...") {
+			s.T().Log("Successfully flushed server and reset aggregators.")
+		}
+	}, 1*time.Minute, 10*time.Second)
+
+	// Create a new log file for utf encoded messages
+	s.Env().RemoteHost.MustExecute("sudo touch /var/log/hello-world-utf.log")
+	s.Env().RemoteHost.MustExecute("sudo chmod +r /var/log/hello-world-utf.log")
+}
+
+func (s *UtfSuite) AfterTest(suiteName, testName string) {
+	s.BaseSuite.AfterTest(suiteName, testName)
+
+	// flush intake
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		err := s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+		if assert.NoErrorf(c, err, "Having issues flushing server and resetting aggregators, retrying...") {
+			s.T().Log("Successfully flushed server and reset aggregators.")
+		}
+	}, 1*time.Minute, 10*time.Second)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why do you flush after test ? Since you `flush` before tests, that should be enough to ensure the next test runs on a flushed fakeintake